src/components/pages/LoginPage.vue

```diff
@@ -0,0 +1,172 @@
+  components: {
+    TheContainer,
+    AppInput,
+    AppButton,
+    FormCard,
+    SectionSubtitle,
+    SectionTitle,
+    ValidationMessage
+  },
+  setup() {
+    const auth = useAuthStore();
+    return { auth };
+  },
+  data() {
+    return {
+      email: '',
+      password: '',
+      emailTouched: false,
+      passwordTouched: false,
+    }
+  },
+  computed: {
+    isLoggedIn() {
+      return this.auth.isLoggedIn
+    },
+    emailErrorMessage() {
+      if (this.email === '' && this.emailTouched) {
+        return 'Please enter your email'
+      }
+      return ''
+    },
+    passwordErrorMessage() {
+      if (this.password === '' && this.passwordTouched) {
+        return 'Please enter your password'
+      }
+      return ''
+    },
+    isFormValid() {
+      return this.emailProvided && this.passwordProvided
+    }
+  },
+  methods: {
+    allTouched() {
+      this.emailTouched = true;
+      this.passwordTouched = true;
+    },
+    async handleLogin(e) {
+      e.preventDefault();
+      this.allTouched();
+      try {
+        await this.auth.login({
+          email: this.email,
+          password: this.password
+        })
+        this.$router.push({name: 'ChooseSeats'})
+      } catch (error) {
+        console.error(error)
+      }
+    }
+  }
+})
+</script>
+
+<template>
+  <div class="login-page">
+    <TheContainer variant="form">
+      <div class="login-page__header">
+        <SectionTitle size="large">Hi there!</SectionTitle>
+        <SectionSubtitle size="large">Care to log in?</SectionSubtitle>
+      </div>
+      <FormCard>
+        <form
+          @submit="handleSubmit"
+        >
+          <AppInput
+            label="e-mail"
+            type="email"
+            placeholder="Something ending with monterail.com"
+            v-model="email"
+            @blur="emailTouched = true"
+            :is-valid="!emailErrorMessage"
+          />
+          <ValidationMessage
+            v-if="emailErrorMessage"
+            class='not-valid'
+          >
+            {{ emailErrorMessage }}
+          </ValidationMessage>
+          <AppInput
+            label="password"
+            type="password"
+            placeholder="Enter your password"
+            v-model="password"
+            @blur="passwordTouched = true"
+            :is-valid="!passwordErrorMessage"
+          />
+          <ValidationMessage
+            v-if="passwordErrorMessage"
+            class='not-valid'
+          >
+            {{ passwordErrorMessage }}
+          </ValidationMessage>
+          <div class="login-page__buttons">
+            <AppButton
+              size="large"
+              color-scheme="no-border"
+              usage="form"
+              :to="{name: 'Register'}"
+            >
+              Register instead
+            </AppButton>
+            <AppButton
+              size="large"
+              color-scheme="main"
+              usage="form"
+              @click="handleLogin"
```

Comment:
Instead of defining a `@click` event here, we should handle the form submission on the form's `@submit` event above, otherwise our form won't be accessible.

Reply:
This and the previous issue were due to a bug in my AppButton component -- it wasn't assigning the type properly so the form wasn't reacting on button clicks. I haven't noticed it and tried to resolve the problem of missing submission with this `@click` on the button. But now it's fixed and works as it should 😌